Add a governance extension that implements super quorum (#5492)
OpenZeppelin/openzeppelin-contracts: .changeset/fuzzy-crews-poke.md

---
'openzeppelin-solidity': minor
---

> `GovernorSuperQuorum`: Add a governance extension to support a super quorum. Proposals that meet the super quorum can be executed earlier than the proposal deadline.

Reviewer: ```suggestion
`GovernorSuperQuorum`: Add a governance extension to support a super quorum. Proposals that meet the super quorum can be queued earlier than the proposal deadline.
```
Not necessarily accurate as is given that the timelock delay still applies.

Reviewer: ```suggestion
`GovernorSuperQuorum`: Add a governance extension to support a super quorum. Proposals that meet the super quorum advance to the `Succeeded` state before the proposal deadline.
```

What about this?

Reviewer 2: actually, none of the proposal are correct. When the super quorum is met, it advances to the succedded state only if the voting has a majority of for.

Reviewer 2: ```suggestion
`GovernorSuperQuorum`: Add a governance extension to support a super quorum. Proposals that meet the super quorum (and have a majority of for votes) advance to the `Succeeded` state before the proposal deadline.
```

Reviewer: Ok good. Let's change all short descriptions of the contract to this.

---
'openzeppelin-solidity': minor
---

- `GovernorSuperQuorum`: Add a governance extension to support a super quorum. Proposals that meet the super quorum can be executed earlier than the proposal deadline.

> - `GovernorVotesSuperQuorumFraction`: Add a governance extension to enable super quorum with fractional voting.

Author: Addressing in https://github.com/OpenZeppelin/openzeppelin-contracts/pull/5492/commits/bb9a98976fca6bf33b7c72c543e9c7ef9fef2925
